index.js

```diff
@@ -240,7 +242,9 @@ var getResult = (errorResourceReq, results, isBib, resourceId, operation, attemp
       var errorInJson = JSON.parse(errorResourceReq)
       log.error('Error httpstatus -' + errorInJson.httpStatus + '-')
       if (errorInJson.httpStatus === 401) {
-        log.error('This is a token issue. Going to renew token')
+        __accessToken = null
+        log.error('This is a token issue')
+        reject(errorResourceReq)
         if (isBib) {
```

Comment:
I'd advise against running code after calling either `resolve` or `reject`. Those are callback functions that conventionally signal the work is done. Here you're rejecting the request because you've determined the token is invalid, but then you immediately call `getWrapperAccessToken` to create a new token. The rejection will likely bubble up to fail the lambda invocation. It's unclear if your `getWrapperAccessToken` will succeed before the lambda environment kills the process. If your intent is to null out the access token and fail the lambda so that it can be regenerated when the event is retried (which is a elegant pattern, I think), I think you should do nothing after rejecting. In general I think you should treat `resolve` and `reject` as `return`s - the last thing you do in a function.